polynote-frontend/polynote/state/notebook_state.ts

```diff
@@ -247,7 +247,7 @@ export class NotebookStateHandler extends BaseHandler<NotebookState> {
                 }
             }
             const currentCell = state.cells[currentCellId];
-            anchor = {id: currentCellId, language: currentCell?.language ?? 'scala', metadata: currentCell?.metadata ?? new CellMetadata()};
+            anchor = {id: currentCellId, language: currentCell?.language && currentCell.language === 'viz' ? 'scala' : currentCell.language, metadata: currentCell?.metadata ?? new CellMetadata()};
```

Comment:
Looks like maybe this should be `... ? 'scala' : (currentCell?.language ?? 'scala')`?

Reply:
How about `(currentCell?.language === undefined || currentCell?.language === 'viz') ? 'scala' : currentCell.language` ?